haystack/nodes/prompt/invocation_layer/amazon_bedrock.py

```diff
@@ -0,0 +1,308 @@
+
+        resize_info = self.prompt_handler(prompt)
+        if resize_info["prompt_length"] != resize_info["new_prompt_length"]:
+            logger.warning(
+                "The prompt has been truncated from %s tokens to %s tokens so that the prompt length and "
+                "answer length (%s tokens) fit within the max token limit (%s tokens). "
+                "Shorten the prompt to prevent it from being cut off.",
+                resize_info["prompt_length"],
+                max(0, resize_info["model_max_length"] - resize_info["max_length"]),  # type: ignore
+                resize_info["max_length"],
+                resize_info["model_max_length"],
+            )
+        return str(resize_info["resized_prompt"])
+
+    @classmethod
+    def supports(cls, model_name_or_path, **kwargs):
+        supported_model_ids = cls.SUPPORTED_MODELS.keys()
+        model_supported = model_name_or_path in supported_model_ids
+        if not model_supported or not cls.aws_configured(**kwargs):
+            return False
+
+        try:
+            session = cls.get_aws_session(**kwargs)
+            bedrock = session.client("bedrock")
+            foundation_models_response = bedrock.list_foundation_models(byOutputModality="TEXT")
+            available_model_ids = [entry["modelId"] for entry in foundation_models_response.get("modelSummaries", [])]
+        except AWSConfigurationError as e:
+            raise AmazonBedrockConfigurationError(message=e.message) from e
+        except Exception as e:
+            raise AmazonBedrockConfigurationError(
+                "Could not connect to Amazon Bedrock. Make sure the AWS environment is configured correctly."
+            ) from e
+
+        model_available = model_name_or_path in available_model_ids
+        if not model_available:
+            raise AmazonBedrockConfigurationError(
+                f"The model {model_name_or_path} is not available in Amazon Bedrock. "
+                f"Please make sure the model is available in the configured AWS region and you've been granted access."
+            )
+
+        return model_supported
+
+    def supports_streaming(self, model_name_or_path):
+        foundation_models_response = self.bedrock.list_foundation_models(byOutputModality="TEXT")
+        available_model_ids = [
+            entry["modelId"]
+            for entry in foundation_models_response.get("modelSummaries", [])
+            if entry.pop("responseStreamingSupported", None)
+        ]
+        if model_name_or_path not in available_model_ids:
+            raise AmazonBedrockConfigurationError(f"{model_name_or_path} does not offer streaming support")
+
+    def invoke(self, *args, **kwargs):
+        kwargs = kwargs.copy()
+        prompt: Any = kwargs.pop("prompt", None)
+        stream: Any = self.model_input_kwargs.pop("stream", None)
+        self.supports_streaming(self.model_name_or_path)
```

Comment:
Calling list_foundation_models on any `invoke` is not ideal. Let's better only check for streaming support in `supports`. If someone passes an inference param `stream` with non streamable models, it's fine to break during `invoke`.